### ISSUE TITLE
Measure transaction and messaging throughput

### ISSUE THREAD

**Issue by Sophia-Gold** [paritytech/polkadot#7065]
As part of marketing asynchronous backing, as well as for general load testing purposes, we'd like to be able to make claims about the increase in TPS and possibly total parachain TPS as well as cross-chain message throughput. This will involve several steps:

- [ ] Develop a transaction generator that can fill a transaction-only (no smart contracts) parachain with transactions and cross-chain messages
- [ ] Test the number of parachains we can support with full blocks and synchronous backing. We'd like this to be 100+, which will likely require some performance tuning, possibly aided by an increase in hardware specs (https://forum.parity.io/t/investigating-new-benchmarking-hardware/687/124).
- [ ] Measure total parachain TPS with synchronous backing and then with asynchronous backing
- [ ] Measure total messaging throughput. It's an open question how to combine this with transaction load, and probably less of a useful metric for marketing, but good for us to know anyway.

**Comment by bkchr** [comment]
Yes, we should no re-invent the wheel again. TPS are rather useless in general, but when we want to do this we should use the mentioned STPS.

**Comment by Sophia-Gold** [comment]
> At some point, we'll do XCMP so if you talk about XCMP bandwidth, then be aware or mention that it should be a pale shadow of what XCMP should do.

I think the messaging throughput part is more just for load testing, and possibly to make a claim about XCMP vs. HRMP in the future. 

What I'm not certain on is whether to include cross-chain messages when measuring TPS. I tend to think we shouldn't.

**Comment by bkchr** [comment]
> What I'm not certain on is whether to include cross-chain messages when measuring TPS. I tend to think we shouldn't.

Sounds reasonable to not include this! Especially as we currently need to send all messages via the relay chain.
